pkg/preflights/host-preflight.yaml

```diff
@@ -367,17 +371,30 @@ spec:
         regex: 'nameserver\s*(localhost|127\.0\.0\.1)'
         outcomes:
           - fail:
-              when: "true"
+              when: 'true'
               message: "Neither 'nameserver localhost' nor 'nameserver 127.0.0.1' can be present in resolv.conf. Remove them to continue."
           - pass:
-              when: "false"
+              when: 'false'
               message: "Neither 'nameserver localhost' nor 'nameserver 127.0.01' is present in resolv.conf"
     - filesystemPerformance:
         checkName: Filesystem Write Latency
         collectorName: filesystem-write-latency-etcd
         outcomes:
           - pass:
-              when: "p99 < 10ms"
+              when: 'p99 < 10ms'
               message: 'P99 write latency for the disk at /var/lib/k0s/etcd is {{ "{{" }} .P99 {{ "}}" }}, which is better than the 10 ms requirement.'
           - fail:
               message: 'P99 write latency for the disk at /var/lib/k0s/etcd is {{ "{{" }} .P99 {{ "}}" }}, but it must be less than 10 ms. A higher-performance disk is required.'
+    - jsonCompare:
+        checkName: Wildcard DNS
+        fileName: host-collectors/dns/wildcard-check/result.json
+        path: 'resolvedFromSearch'
+        value: |
+          ""
+        outcomes:
+          - fail:
+              when: 'false'
+              message: 'Possible wildcard DNS entry detected at: {{ "{{" }}  .resolvedFromSearch {{ "}}" }} . Remove the search domain OR remove the wildcard DNS entry.'
```

Comment:
I'm not going to write a code suggestion because I can't test it easily, but here is my desired output (open to suggestions too):
>Wildcard DNS entry *.foo.testcluster.net detected. Remove the wildcard DNS entry or the foo.testcluster.net search domain from resolv.conf.